slither/core/dominators/utils.py

```diff
@@ -95,5 +95,4 @@ def compute_dominance_frontier(nodes: List["Node"]) -> None:
                     runner.dominance_frontier = runner.dominance_frontier.union({node})
                 while runner != node.immediate_dominator:
                     runner.dominance_frontier = runner.dominance_frontier.union({node})
-                    assert runner.immediate_dominator
```

Comment:
For the sake of the type annotation, should we do a break if the immediate dominator is none?

This will also allow us to add a code comment regarding the dead code